service/tbc/tbc.go

```diff
@@ -1677,6 +1878,58 @@ func (s *Server) FeesAtHeight(ctx context.Context, height, count int64) (uint64,
 	return fees, errors.New("not yet")
 }
 
+// FullBlockAvailable returns whether TBC has the full block
+// corresponding to the specified hash available in its database.
+func (s *Server) FullBlockAvailable(ctx context.Context, hash *chainhash.Hash) (bool, error) {
+	if s.cfg.ExternalHeaderMode {
+		return false, errors.New("cannot call full block available on TBC running in External Header mode")
+	}
+
+	block, err := s.db.BlockByHash(ctx, hash)
+	if err != nil {
+		if errors.Is(err, database.ErrBlockNotFound) {
+			return false, nil // Not found, but not an error to pass upstream
+		} else {
+			return false, err // Another error was encountered which upstream should know about
+		}
+	}
+
+	if block != nil {
+		// Were able to get full block from database
+		return true, nil
+	} else {
+		return false, errors.New("fetching block did not return error but block is nil")
+	}
+}
```

Comment:
This should return an actionable error instead of mixing flags and bools creating 4 actionable cases.